### PR TITLE
Feature/scat 3359 score and comment

### PR DESCRIPTION
**Pull request by venkibric**
### JIRA link (if applicable) ###

SCAT-3359

### Change description ###

Added Score and Comment controller

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java [hunk 1]
```diff
@@ -108,143 +88,20 @@ public String sendOrRespondMessage(final String profile, final Integer projectId
       if (CollectionUtils.isEmpty(nonOCDS.getReceiverList())) {
         throw new JaggaerRPAException("Suppliers are mandatory if broadcast is 'No'");
       }
-      var supplierString = validateSuppliers(procurementEvent, nonOCDS);
+      var suppliers = rpaGenericService.getValidSuppliers(procurementEvent,
+          nonOCDS.getReceiverList().stream().map(org -> org.getId()).collect(Collectors.toList()));
+      var appendSupplierList = new StringBuilder();
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java:93]: Use `StringJoiner` for simplicity

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java [hunk 2]
```diff
@@ -108,143 +88,20 @@ public String sendOrRespondMessage(final String profile, final Integer projectId
       if (CollectionUtils.isEmpty(nonOCDS.getReceiverList())) {
         throw new JaggaerRPAException("Suppliers are mandatory if broadcast is 'No'");
       }
-      var supplierString = validateSuppliers(procurementEvent, nonOCDS);
+      var suppliers = rpaGenericService.getValidSuppliers(procurementEvent,
+          nonOCDS.getReceiverList().stream().map(org -> org.getId()).collect(Collectors.toList()));
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/MessageService.java:92]: Sonar suggests replacing this with a method reference (i.e. `OrganizationReference1::getId`)

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/SupplierService.java [hunk 3]
```diff
@@ -64,20 +81,72 @@
...
+        procurementEvent, scoreAndComments.stream().map(e -> e.getOrganisationId()).collect(Collectors.toList()));
+    var appendSupplierList = new StringBuilder();
+    var appendScoreList = new StringBuilder();
+    var appendCommentList = new StringBuilder();
+
+    for (Supplier supplier : validSuppliers.getFirst()) {
+      String orgId = validSuppliers.getSecond().stream()
+          .filter(e -> e.getExternalOrganisationId().equals(supplier.getCompanyData().getId())).findFirst().get()
+          .getOrganisationId();
+      ScoreAndCommentNonOCDS scoreAndCommentNonOCDS = scoreAndCommentMap.get(orgId);
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/SupplierService.java:125]: Can replace `ScoreAndCommentNonOCDS` with `var` (Java 11 Sonar warning)

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/SupplierService.java [hunk 4]
```diff
@@ -64,20 +81,72 @@
...
+      appendSupplierList.append(supplier.getCompanyData().getName());
+      appendSupplierList.append("~|");
+      appendScoreList.append(scoreAndCommentNonOCDS.getScore());
+      appendScoreList.append("~|");
+      appendCommentList.append(scoreAndCommentNonOCDS.getComment());
+      appendCommentList.append("~|");
+    }
+
+    // input details
+    String suppliers = appendSupplierList.toString().substring(0, appendSupplierList.toString().length() - 2);
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/SupplierService.java:135]: Can replace each of these Strings with `var` (Java 11 Sonar warning)

**Review by tommyb82 — APPROVED** [review]
Looks good now Venki, happy for it to go to dev if you are 👍

**Pull request merged**